sql: flake in TestTenantLogic_alter_table/column_backfiller_update_batching

Here: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelEssentialCi/9340633?showRootCauses=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true
```
Failed
=== RUN   TestTenantLogic_alter_table/column_backfiller_update_batching
    logic.go:2887: let $use_decl_sc = on
    logic.go:2841:
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1768/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/alter_table:2294: SELECT count(*) FROM tb WHERE v = 'abc';
        expected success, but found
        (40001) restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_ASYNC_WRITE_FAILURE - missing intent on: /Tenant/10/Table/199/1/852336628213547009/0): "sql txn" meta={id=93b3a5fd key=/Tenant/10/Table/7/1/0/0 pri=0.01595430 epo=0 ts=1680182896.449159057,1 min=1680182896.390096661,0 seq=0} lock=true stat=PENDING rts=1680182896.390096661,0 wto=false gul=1680182896.890096661,0

[13:28:19] --- total progress: 470 statements
--- total: 470 tests, 2 failures
    --- FAIL: TestTenantLogic_alter_table/column_backfiller_update_batching (0.29s)
```

Jira issue: CRDB-26257

--- a/pkg/sql/logictest/testdata/logic_test/alter_primary_key
+++ b/pkg/sql/logictest/testdata/logic_test/alter_primary_key
@@ -1005,6 +1005,8 @@ SELECT index_id, index_name FROM crdb_internal.table_indexes WHERE descriptor_na
 statement ok
 DROP TABLE IF EXISTS t
 
+skip_on_retry
+
 statement ok
 BEGIN
 

--- a/pkg/sql/logictest/testdata/logic_test/alter_table
+++ b/pkg/sql/logictest/testdata/logic_test/alter_table
@@ -2245,6 +2245,8 @@ SELECT * FROM multipleinstmt ORDER BY id ASC;
 2  b  b  false  NULL  true  NULL
 3  c  c  false  NULL  true  NULL
 
+skip_on_retry
+
 subtest column_backfiller_update_batching
 
 let $use_decl_sc

--- a/pkg/sql/logictest/testdata/logic_test/schema_change_in_txn
+++ b/pkg/sql/logictest/testdata/logic_test/schema_change_in_txn
@@ -683,6 +683,8 @@ subtest rollback_mutations
 statement ok
 INSERT INTO customers (k) VALUES ('z'), ('x')
 
+skip_on_retry
+
 statement ok
 BEGIN
 

